libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -515,22 +516,41 @@ - (void)resumeSyncUpOneEntry:(SFSyncState*)sync
     };
     
     // Create handler
-    SFSyncServerTargetCompleteBlock completeBlockCreate = ^(NSDictionary *d) {
+    SFSyncUpTargetCompleteBlock completeBlockCreate = ^(NSDictionary *d) {
         // Replace id with server id during create
         record[kId] = d[kSyncManagerLObjectId];
         completeBlockUpdate(d);
     };
     
-    SFSyncServerTargetCompleteBlock completeBlock;
+    
+    SFSyncUpTarget *target = (SFSyncUpTarget*) sync.target;
+    NSArray *fieldList = sync.options.fieldlist;
+    
+    // Getting type and id
+    NSString* objectType = [SFJsonUtils projectIntoJson:record path:kObjectTypeField];
+    NSString* objectId = record[kId];
+    
+    // Fields to save (in the case of create or update)
+    NSMutableDictionary* fields = [NSMutableDictionary dictionary];
```

Comment:
Agreed it would be more flexible but given that the sync up is parameterised via the fieldlist (passed through the sync options), won't we simply end up with the same code in all custom targets (looping through the fieldlist, and projecting out the chosen fields)? Also, if you really need the whole record, you could simply pass [""] for the field list and you end up with {"":record} in fields?